### PR TITLE
[NOD-679] Use transaction hash instead of id for transaction uniqueness

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: dbaccess/transaction.go [hunk 1]
```diff
@@ -161,6 +161,81 @@ func AcceptedTransactionsByBlockID(ctx Context, blockID uint64, preloadedFields
 	return transactions, nil
 }
 
+// TransactionsByHashes retrieves all transactions by their `transactionHashes`.
+// If preloadedFields was provided - preloads the requested fields
+func TransactionsByHashes(ctx Context, transactionHashes []string, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Transaction, error) {
```
Comment (svarogg) [inline dbaccess/transaction.go]: Please break line

File: dbaccess/transaction.go [hunk 2]
```diff
@@ -161,6 +161,81 @@ func AcceptedTransactionsByBlockID(ctx Context, blockID uint64, preloadedFields
...
+		return nil, httpserverutils.NewErrorFromDBErrors("some errors were encountered when loading transactions from the database:", dbErrors)
+	}
+
+	return txs, nil
+}
+
+// TransactionsByIDsAndBlockID retrieves all transactions in a
+// block with the given ID by their `transactionIDs`.
+// If preloadedFields was provided - preloads the requested fields
+func TransactionsByIDsAndBlockID(ctx Context, transactionIDs []string, blockID uint64, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Transaction, error) {
```
Comment (svarogg) [inline dbaccess/transaction.go]: Please break line

File: dbaccess/transaction_output.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ func UTXOsByAddress(ctx Context, address string, preloadedFields ...dbmodels.Fie
 	query := db.
 		Joins("LEFT JOIN `addresses` ON `addresses`.`id` = `transaction_outputs`.`address_id`").
 		Joins("INNER JOIN `transactions` ON `transaction_outputs`.`transaction_id` = `transactions`.`id`").
-		Where("`addresses`.`address` = ? AND `transaction_outputs`.`is_spent` = 0", address)
+		Where("`transactions`.`accepting_block_id` IS NOT NULL AND  `addresses`.`address` = ? AND `transaction_outputs`.`is_spent` = 0", address)
```
Comment (svarogg) [inline dbaccess/transaction_output.go]: break line plx

**Review by svarogg — CHANGES_REQUESTED** [review]

**Review by svarogg — DISMISSED** [review]

File: kasparovsyncd/sync/block.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ func insertBlocks(dbTx *dbaccess.TxContext, blocks []*rawAndVerboseBlock, transa
 	for i, block := range blocks {
 		blockMass := uint64(0)
 		for _, tx := range block.Verbose.RawTx {
-			blockMass += transactionIDsToTxsWithMetadata[tx.TxID].mass
+			blockMass += transactionIDsToTxsWithMetadata[tx.Hash].mass
```
Comment (dereeno) [inline kasparovsyncd/sync/block.go]: Please rename the mapping to `transactionHashesToTxsWithMetadata`

**Review by dereeno — CHANGES_REQUESTED** [review]

**Review by dereeno — APPROVED** [review]

**Review by svarogg — APPROVED** [review]

**Pull request merged**